### ISSUE TITLE
Volume adjustment is broken

### ISSUE THREAD

**Issue by RobinGiel** [benwiley4000/cassette#407]
seeing this in the console:

cassette-player.js:653 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

Having this problem with the volume bar that I cannot set the volume. The volume bar either goes to mute or on full volume.  It is suggested to add a passive handler.

**Comment by benwiley4000** [comment]
@RobinGiel thanks for flagging! I think I'm having this issue as well, although I don't think the message in the console is exactly related. What we *want* is a non-passive listener. However it might be adding a passive event listener to the progress bar now... I might need to go through and add `passive: false` to all the listeners for `touchstart`.

**Comment by benwiley4000** [comment]
Turns out the message you saw had nothing to do with the actual problem, which is that my performance refactor for the progress bar meant we had stopped correctly monitoring changes to the progress bar's bounding box. Since the volume bar first renders hidden, it had a 0x0 bounding box, and never updated.

Fixed in v2.0.0-alpha.33.

**Comment by RobinGiel** [comment]
Hi @benwiley4000, I noticed that the volume button is working fine inside a flexbox or div, or within a page with custom height and scrollbar. On the documentation page I noticed it's working just fine and that's becuase of this css: div[class*='rsg--hasSidebar'] {
    overflow: scroll;
    height: calc(100vh - 50px);
    min-height: initial;

So I went and use the exact same Css for my website and it works just fine. 
The problem is, if I decide to use it with the native scroll the problem still remains. 
If I use:
    overflow: scroll;
    height: calc(100vh - 50px)

I'm am very limited using other plugins with the native scroll

**Comment by benwiley4000** [comment]
Thanks for flagging! Would it be easy for you to create a repro of this?

**Comment by benwiley4000** [comment]
Hey @RobinGiel if I'm following correctly, the correct way to repro your problem is to remove the `overflow: scroll` from that class? I did and it took me back to a similar volume adjustment problem as before.

It's sort of hard for me to understand at the moment what is wrong. I'm going to reopen this issue since it's a similar sort of problem, even if the cause may be different.

**Comment by RobinGiel** [comment]
Hi @benwiley4000, while testing it on other browsers i can confirm that this bug is not happening in Safari

**Comment by RobinGiel** [comment]
@benwiley4000 I finally found the bug. instead of using:

 `function updateScroll() {
  scrollTop = document.body.scrollTop;
  scrollLeft = document.body.scrollLeft;
}`


I use: `function updateScroll() {
  scrollTop = window.scrollY;
  scrollLeft = window.scrollX;
}`

and this solves the issue.

following advices here: 

[https://stackoverflow.com/questions/19635188/why-is-body-scrolltop-deprecated#answer-49699592 ](url)

[https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollX](url)

[https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollY](url)

**Comment by benwiley4000** [comment]
Thanks @RobinGiel ! What browser version are you running, by curiosity?

Looking at those pages it seems like the best way to be compatible across multiple browsers is:

```js
  scrollTop = window.pageXOffset !== undefined
    ? window.pageXOffset
    : document.body.scrollTop;
  scrollLeft = window.pageYOffset !== undefined
    ? window.pageYOffset
    : document.body.scrollLeft;
```

I think... haven't tested it. Would you like to work on a pull request?

**Comment by RobinGiel** [comment]
Hey @benwiley4000, I'm running the latest version of chrome and safari. According to the MDN docs, window.scrollY and window.scrollX are compatible in all browsers expect internet explorer. 

I will test this and make a pull request soon

**Comment by benwiley4000** [comment]
Thanks! I'm using these aliases because supposedly older versions use those
instead and also I do want to support Internet Explorer 10 and 11 if
possible.

Le lun. 16 sept. 2019 14 h 47, Robin Giel <notifications@github.com> a
écrit :

**Comment by benwiley4000** [comment]
Closed by #430 (I think).
